app/controllers/publish/courses/engineers_teach_physics_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Publish
+  module Courses
+    class EngineersTeachPhysicsController < PublishController
+      decorates_assigned :course
+      include CourseBasicDetailConcern
+
+      def new
+        authorize(@provider, :can_create_course?)
+        return if has_physics_subject?
+
+        redirect_to next_step
+      end
+
+      def edit; end
+
+      def update; end
+
+      def back
+        authorize(@provider, :edit?)
+        if has_physics_subject?
+          redirect_to new_publish_provider_recruitment_cycle_courses_engineers_teach_physics_path(path_params)
+        else
+          redirect_to @back_link_path
+        end
+      end
+
+    private
+
+      def has_physics_subject?
+        @course.master_subject_id == 29
```

Comment:
Can we rely on this? Would it be better to find the subject by its `subject_code` and then assign the id?

Reply:
I am open to best suggestions here

Reply:
But at the moment I think this is the best option